core-js/src/main/javascript/cdf/components/BaseComponent.js

```diff
@@ -513,6 +524,90 @@ define([
       return opts || {};
     },
 
+    /**
+     * @summary Disposes this component, if it wasn't already disposed.
+     * @description Disposes this component, if it wasn't already disposed.
+     */
+    dispose: function() {
+      if(this.isDisposed) {
+        return;
+      }
+
+      this.isDisposed = true;
+
+      if(this.dashboard != null) {
+        this.dashboard.removeComponent(this);
+      } else {
+        // Usually, the whole dashboard is being disposed of.
+        this._releaseQuery();
```

Comment:
Should call `_unlink` here, and it would contain `_releaseQuery` and `_releasePlaceholder`. It would force implementations to call base, though.